From ea050b5974992228f0bb46d8d4108d15880b5120 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Mon, 1 Mar 2021 19:18:22 +0100
Subject: [PATCH] Disable backtrace generation

The backtracing functionality is quite fragile (depends on parsing output
from "nm" and "addr2line", calls signal unsafe functions in signal
handlers).

Also, linking fails due to missing linkage to libdl (uses dladdr).
---
 libsrc/core/exception.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libsrc/core/exception.cpp b/libsrc/core/exception.cpp
index b89d721e..335e3411 100644
--- a/libsrc/core/exception.cpp
+++ b/libsrc/core/exception.cpp
@@ -23,7 +23,7 @@ namespace ngcore
 
 
 // ********* STUFF FOR GETBACKTRACE ***************************
-#ifdef __GNUC__
+#if 0
 
 #include <execinfo.h>
 #include <string.h>
@@ -232,6 +232,10 @@ namespace ngcore
 {
   std::string GetBackTrace()
   {
+    static auto backtrace = getenv("NG_BACKTRACE");
+    if(backtrace) {
+	return std::string("Backtrace disabled, use a debugger");
+    }
     return std::string();
   }
 } // namespace ngcore
-- 
2.30.1

